emeis/core/tests/test_user.py

```diff
@@ -1,6 +1,33 @@
 from django.urls import reverse
 from rest_framework import status
 
+from ..models import User
+
+
+def test_user_model():
+    username_raw = (
```

Comment:
Tests like this might be a really good use case for `hypothesis`. We should use that way more anyhow :)

Reply:
Nice! I'm not familiar with hypothesis yet. How would you approach this in this situation? We want to test with very specific input and expect specific output. We don't want to test `self.normalize_username()`, we just want to make sure it has been called.

Reply:
You could do something like this:

```python
from hypothesis import given
from hypothesis.strategies import text

@given(username=text())
 def test_user_model(username):
    username_normal = unicodedata.normalize("NFKC", username)
    # do the thing
    assert user.username == username_normal
```

This will run your test a couple of times using various usernames (Hypothesis is really good at producing edge cases too!). You could also fill insert email, first and last name from Hypothesis as well :)

Reply:
Nice! Will try!

Reply:
@winged Awesome! Works!

Reply:
Yeah! Very cool!